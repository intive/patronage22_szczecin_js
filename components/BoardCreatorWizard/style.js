import styled from 'styled-components'

export const Subtitle = styled.div`
  margin-top: 0.75rem;
  font-size: 0.75rem;
  line-height: 1.5rem;
  text-align: center;
  color: #7E7E7E;
`

export const TextInput = styled.input`
  border: none;
  border-bottom: 1px solid #7E7E7E;
  outline: none;
  width: 100%;
  padding: 1rem;
  margin-top: 1rem;
  font-size: 1.5rem;
  caret-color: #515151;
  color: #515151;
}
`
