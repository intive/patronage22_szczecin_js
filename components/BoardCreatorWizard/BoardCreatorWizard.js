import { Subtitle, TextInput, CharacterCounterStyled } from './style'
import { useTranslation } from 'next-i18next'

const BoardCreatorWizard = ({ onChange, maxLength, value }) => {
  const { t } = useTranslation('common')
  return (
    <>
      <Subtitle>{t('createNewBoard.createNewBoardSubtitle')}</Subtitle>
      <TextInput autoFocus type='text' onChange={onChange} maxLength={maxLength} />
      {maxLength ? <CharacterCounterStyled inputValue={value} maxCharacters={maxLength} /> : null}
    </>
  )
}

export default BoardCreatorWizard
