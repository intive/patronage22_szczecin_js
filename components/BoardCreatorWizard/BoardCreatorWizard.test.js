/* globals describe, it, expect  */
/* eslint-env jest */

import { render, screen } from '@testing-library/react'
import BoardCreatorWizard from './BoardCreatorWizard'

describe('BoardCreatorWizard', () => {
  it('renders boardCreatorWizard', () => {
    const { container } = render(
      <BoardCreatorWizard />
    )

    expect(container).toMatchSnapshot()
  })

  it('should filled input field correctly', () => {
    const inputData = { boardname: 'My new board' }
    const { container } = render(
      <BoardCreatorWizard />
    )
    const boardnameInput = screen.getByRole('textbox')
    boardnameInput.value = inputData.boardname

    expect(boardnameInput.value.length).toBe(12)
    expect(boardnameInput.value).toBe('My new board')
    expect(container).toMatchSnapshot()
  })
})
