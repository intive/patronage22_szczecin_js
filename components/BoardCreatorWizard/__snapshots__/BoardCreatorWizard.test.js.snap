// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardCreatorWizard renders boardCreatorWizard 1`] = `
<div>
  <div
    class="sc-bdvvtL lgusKh"
  >
    createNewBoard.createNewBoardSubtitle
  </div>
  <input
    class="sc-gsDKAQ eZwtwJ"
    type="text"
  />
</div>
`;

exports[`BoardCreatorWizard should filled input field correctly 1`] = `
<div>
  <div
    class="sc-bdvvtL lgusKh"
  >
    createNewBoard.createNewBoardSubtitle
  </div>
  <input
    class="sc-gsDKAQ eZwtwJ"
    type="text"
  />
</div>
`;
